omero/developers/json-api.txt

```diff
@@ -0,0 +1,460 @@
+    normalize   Boolean     Place Experimenters and Groups into top-level lists instead
+                            of nesting within objects
+    childCount  Boolean     Use ?childCount=true to include an omero:childCount attribute
+                            for container objects
+    owner       Number      Filter by Experimenter ID
+
+    group       Number      Filter by Group ID
+
+
+Error handling
+^^^^^^^^^^^^^^
+
+Invalid parameters or invalid JSON will result in a ```400 Bad Request```.
+For example, ```?limit=foo``` will give a response of:
+
+::
+
+    {"message": "invalid literal for int() with base 10: 'foo'"}
+
+
+Unhandled exceptions are handled with a ```500``` error response with an
+error "message" and may also include a "stactrace".
+
+
+Getting Started
+---------------
+
+
+List supported versions
+^^^^^^^^^^^^^^^^^^^^^^^
+
+You need to find which versions of the API are supported by your server,
+as described above. These are provided by the base url:
+
+::
+
+    GET     /api/
+
+
+**Response**
+
+::
+
+    {
+      "data": [
+        {
+          "version": "0",
+          "url:base": "http://<server>/api/v0/"
+        }
+      ]
+    }
+
+
+List starting urls
+^^^^^^^^^^^^^^^^^^
+
+The base url for the chosen version will list a number of urls for
+logging on and getting started.
+
+::
+
+    GET     /api/v0/
+
+
+**Response**
+
+::
+
+    {
+      "url:login": "http://<server>/api/v0/login/",
+      "url:save": "http://<server>/api/v0/m/save/",
+      "url:projects": "http://<server>/api/v0/m/projects/",
+      "url:plates": "http://<server>/api/v0/m/plates/",
+      "url:datasets": "http://<server>/api/v0/m/datasets/",
+      "url:token": "http://<server>/api/v0/token/",
+      "url:schema": "http://www.openmicroscopy.org/Schemas/OME/2016-06",
+      "url:screens": "http://<server>/api/v0/m/screens/",
+      "url:servers": "http://<server>/api/v0/servers/",
+      "url:images": "http://<server>/api/v0/m/images/"
+    }
+
+
+List available OMERO servers
+----------------------------
+
+Your API may allow you to connect to several different OMERO servers.
+
+::
+
+    GET     /api/v0/servers/
+
+
+**Response**
+
+::
+
+
+
+    {
+      "data": [
+        {
+          "host": "<server>",
+          "server": "omero",
+          "id": 1,
+          "port": 4064
+        }
+      ]
+    }
+
+
+Get CSRF token
+--------------
+
+In order to prevent CSRF attacks, will need to obtain a CSRF token
+for your session and use it for all subsequent POST, PUT and DELETE
+requests in that session. You can obtain this from the ```csrftoken```
+cookie of any request or from the following request:
+
+::
+
+    GET     /api/v0/token/
+
+
+**Response**
+
+::
+
+    {
+      "data": "eNoVq528bOqlhQqbCzKuviODTRX3PUO2"
+    }
+
+
+Login
+-----
+
+You can login to create an OMERO session. You must also include the CSRF token.
+The EventContext for this session will be returned to you.
+
+    POST    /api/v0/login/
```

Comment:
This is indented more than the above examples, giving purple highlighting in plone